lnd_test.go

```diff
@@ -2790,6 +2790,58 @@ func testGraphTopologyNotifications(net *networkHarness, t *harnessTest) {
 	ctxt, _ = context.WithTimeout(context.Background(), timeout)
 	closeChannelAndAssert(ctxt, t, net, net.Bob, chanPoint, false)
 
+	// Finally, we will test that when a node starts up, a new node announcement
+	// is only propagated through the network when configuration options have
+	// changed the node announcement since the last instance of the node.
+
+	// First restart Bob with no new config changes
+	if err := net.RestartNode(net.Bob, nil); err != nil {
+		t.Fatalf("unable to restart Bob's node: %v", err)
+	}
+
+	if err := net.ConnectNodes(ctxb, net.Alice, net.Bob); err != nil {
+		t.Fatalf("unable to connect alice to bob: %v", err)
+	}
+
+	// Now restart Bob and add a new IP address to its config options
+	if err := net.RestartNode(net.Bob, func() error {
+		net.Bob.cfg.ExternalIPs = []string{"127.0.0.1"}
+		return nil
+	}); err != nil {
+		t.Fatalf("unable to restart Bob's node: %v", err)
+	}
+
+	if err := net.ConnectNodes(ctxb, net.Alice, net.Bob); err != nil {
+		t.Fatalf("unable to connect alice to bob: %v", err)
+	}
+
+	// Even though we restarted Bob twice, we only expect one node announcement
+	// from when Bob was restarted with config changes.
+	numBobAnnRecv := 0
+	for i := 0; i < 2; i++ {
+		select {
+		case graphUpdate := <-graphUpdates:
+
+			if len(graphUpdate.NodeUpdates) > 0 {
+				nodeUpdate := graphUpdate.NodeUpdates[0]
+				switch nodeUpdate.IdentityKey {
+				case net.Bob.PubKeyStr:
+					numBobAnnRecv++
+					if numBobAnnRecv > 1 {
+						t.Fatalf("unexpected amount of node updates received from Bob")
+					}
+				default:
+					t.Fatalf("unknown node update pubey: %v",
+						nodeUpdate.IdentityKey)
+				}
+			}
+		case <-time.After(time.Second * 10):
```

Comment:
Will this block for 10 seconds also in the success case? I think it is better to check that we get the expected announcement then do a quick wait (~1 sec?) to make sure we are not getting any more.

Reply:
@halseth not that we require longer timeouts when running on Travis due to the jitter and slowness of the machines.